pkg/state/impl/etcd/key.go

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package etcd
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/url"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+)
+
+const (
+	etcdKeyPrefix = "cosi"
+)
+
+// etcdKeyFromPointer builds a key for the resource with the given pointer.
+func (st *State) etcdKeyFromPointer(pointer resource.Pointer) string {
+	prefix := etcdKeyPrefixFromKind(pointer)
+
+	saltedID := fmt.Sprintf("%s%s", pointer.ID(), st.salt)
+	hashedID := sha256hex(saltedID)
+
+	return fmt.Sprintf("%s%s", prefix, hashedID)
+}
+
+// etcdKeyPrefixFromKind returns a key prefix for the given resource kind.
+func etcdKeyPrefixFromKind(kind resource.Kind) string {
+	nsEscaped := url.QueryEscape(kind.Namespace())
+	typeEscaped := url.QueryEscape(kind.Type())
+
+	return fmt.Sprintf("/%s/%s/%s/", etcdKeyPrefix, nsEscaped, typeEscaped)
+}
+
+func sha256hex(input string) string {
```

Comment:
```suggestion
func sha256hex(input []byte) string {
```

I would argue it should accept `[]byte`

Reply:
done